src/main/kotlin/com/github/devcordde/devcordbot/command/impl/CommandClientImpl.kt

```diff
@@ -64,113 +63,80 @@ class CommandClientImpl(
-    private val responsePool = Executors.newSingleThreadExecutor(
-        DefaultThreadFactory("CommandResponseWatcher")
-    ).asCoroutineDispatcher()
-    private val editThreshold = Duration.of(
-        30,
-        ChronoUnit.SECONDS
-    )
 
     override val commandAssociations: MutableMap<String, AbstractCommand> = mutableMapOf()
     override val errorHandler: ErrorHandler = if (bot.debugMode) DebugErrorHandler() else HastebinErrorHandler()
 
-    internal fun acknowledgeResponse(key: Long, channelId: Long, messageId: Long) {
-        messageStorage[key] = channelId to messageId
-        GlobalScope.launch(responsePool) {
-            delay(editThreshold.toMillis())
-            messageStorage.remove(key)
-        }
-    }
-
-
-    /**
-     * Deletes old command response
-     */
-    @EventSubscriber
-    fun onMessageDelete(event: GuildMessageDeleteEvent): Boolean = deleteOldMessage(event.messageIdLong, event.guild)
+    @CheckReturnValue
+    fun updateCommands(): RestAction<Unit> {
+        val commandUpdate = bot.guild.updateCommands()
 
-    /**
-     * Listens for new private messages
-     */
-    @EventSubscriber
-    fun onPrivateMessage(event: DevCordMessageReceivedEvent): Unit = dispatchPrivateMessageCommand(event.message, event)
-
-    private fun dispatchPrivateMessageCommand(message: Message, event: Event) {
-        if (!bot.isInitialized) return
-
-        val author = message.author
-        if (message.isWebhookMessage or author.isBot) return
-
-        bot.guild.getMemberById(author.id) ?: return
-
-        return parseCommand(message, event)
-    }
-
-    /**
-     * Listens for message updates.
-     */
-    @EventSubscriber
-    fun onMessageEdit(event: DevCordGuildMessageEditEvent) {
-        if (Duration.between(event.message.timeCreated, OffsetDateTime.now()) > editThreshold
-        ) return
-        dispatchGuildCommand(event.message, event)
-        deleteOldMessage(event.messageIdLong, event.guild)
+        val commands = commandAssociations.values.distinct().map {
+            when (it) {
+                is AbstractSingleCommand -> it.toSlashCommand()
+                is AbstractRootCommand -> it.toSlashCommand()
+                else -> error("Invalid command: $it")
+            }
+        }
+        commandUpdate.addCommands(commands)
+        return commandUpdate
+            .flatMap { bot.guild.retrieveCommands() }
+            .flatMap {
+                val registeredCommands =
+                    it.map { registeredCommand -> registeredCommand to commandAssociations[registeredCommand.name]!! }
+
+                val actions = registeredCommands.map { (slashCommand, command) ->
+                    slashCommand.updatePermissions(bot.guild.id) {
+                        addAll(command.myPermissions(botOwners, modRole, adminRole))
+                    }
+                }
+
+                RestAction.allOf(actions)
+            }
+            .map { /* Unit */ }
```

Comment:
What exactly is the purpose of this line?

Reply:
It converts to a `RestAction<Unit>`